bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1913,6 +1921,20 @@ public static Image win32_new(Device device, int type, long handle, int nativeZo
 	return new Image(device, type, handle, nativeZoom);
 }
 
+private static class ZoomContext {
+	private int targetZoom;
+	private int nativeZoom;
+
+	private ZoomContext(int targetZoom) {
+		this(targetZoom, targetZoom);
+	}
+
+	private ZoomContext(int targetZoom, int nativeZoom) {
+		this.targetZoom = targetZoom;
+		this.nativeZoom = nativeZoom;
+	}
+}
+
```

Comment:
You could turn this into a record and help the C2 compiler optimize it (no guarantees though)
```suggestion
private static record ZoomContext(int targetZoom, int nativeZoom) {
	ZoomContext(int zoom) {
		this(zoom, zoom);
	}
}
```